### ISSUE TITLE
Adding Google Cloud local SSD as cache drive fails

### ISSUE THREAD

**Issue by MarkGavalda** [zfsonlinux/zfs#4954]
Google Compute Engine offers "local SSDs" that you can attach to your VM on creation via SCSI or NVMe with a theoretical max performance of 680,000 read IOPs and 360,000 write IOPs. Great for using as a cache drive for ZoL pools... if only it worked properly.

The following command on Ubuntu 16.04 4.4.0-34-generic and ZFS v0.6.5.6:
zpool add lxd cache /dev/disk/by-id/google-local-ssd-0

Results in:

```
    PID USER        RES  %CPU %MEM COMMAND                                                                                                                  
   3907 root       0.0m  95.6  0.0 z_null_int                                                                                                               
   3906 root       0.0m  23.2  0.0 z_null_iss
```

on an otherwise idling system!

dmesg is full of:

```
[10574.880773] sd 1:0:1:0: [sdd] Bad block number requested
[10574.887511] sd 1:0:1:0: [sdd] Bad block number requested
[10574.894404] sd 1:0:1:0: [sdd] Bad block number requested
[10574.901222] sd 1:0:1:0: [sdd] Bad block number requested
[10574.908170] sd 1:0:1:0: [sdd] Bad block number requested
[10574.914884] sd 1:0:1:0: [sdd] Bad block number requested
[10574.921717] sd 1:0:1:0: [sdd] Bad block number requested
```

Interestingly using local SSDs for standard pool vdevs works flawlessly.
Once we remove the cache from the pool everything gets back to normal. Also while it's attached the zpool is usable and l2arc seems to work fine other than this.

**Comment by behlendorf** [comment]
@MarkGavalda for these devices what is reported for the physical and logical sector size?  Check `/sys/block/sdc/queue/*_block_size`.  You might also trying forcing the sector alignment to 4k by adding the cache device as follows:

```
zpool add -o ashift=12 pool cache disk
```

**Comment by MarkGavalda** [comment]
Thanks for the tip, unfortunately it didn't help: ashift=12 seems to have no effect.
```
cat /sys/block/sdb/queue/*_block_size
4096
4096
```

**Comment by MarkGavalda** [comment]
Any idea how could we make this usable while waiting for a bug fix @behlendorf ? Thank you!

**Comment by ironMann** [comment]
For starters, could you dump kernel stacks after you add the ssd (Sysrq - l & t)? This might indicate what's going on the system.

**Comment by twarberg** [comment]
Looks like its resolved in 0.7.0-RC5. My guess would be https://github.com/zfsonlinux/zfs/pull/6260

**Comment by behlendorf** [comment]
Closing.  Yes, this should now be resolved.
